upstash_vector/core/index_operations.py

```diff
@@ -64,27 +67,49 @@ def upsert(
             ]
         )
         ```
+
+        #OR 
+
+        ```python
+        from upstash_vector import Vector
+        res = index.upsert(
+            vectors=[
+                DataPayload(id="id5", data="Goodbye-World", metadata={"metadata_f": "metadata_v"}),
+                DataPayload(id="id6", data="Hello-World"),
+            ]
+        )
+        ```
         """
+
         vectors = convert_to_vectors(vectors)
         payload = [
-            {"id": vector.id, "vector": vector.vector, "metadata": vector.metadata}
+            {"id": vector.id, "vector": vector.vector, "metadata": vector.metadata} if isinstance(vector, Vector)
+            else {"id": vector.id, "data": vector.data, "metadata": vector.metadata} 
```

Comment:
If i understand correctly, this code allows us to mix `Vector` and `DataPayload`, which we don't support in the backend

Reply:
added [convert_to_payload](https://github.com/upstash/vector-py/pull/21/files#diff-e08b3898fa8e5022833feee0d3febe4cf89f324b261a1af7fbfc76dd1db7b52aR74) method and [tests](https://github.com/upstash/vector-py/pull/21/files#diff-a6541aa211ac03187d94883cb4ab0a250ec971491fa609f102beb84033108919R449)